lib/chainer/functions/connection/embed_id.rb

```diff
@@ -0,0 +1,52 @@
+module Chainer
+  module Functions
+    module Connection
+      class EmbedIDFunction < Chainer::Function
+        def initialize(ignore_label: nil)
+          @ignore_label = ignore_label
+        end
+
+        def self.embed_id(x, w, ignore_label: nil)
+          self.new(ignore_label: ignore_label).(x, w)
+        end
+
+        def forward(inputs)
+          xm = Chainer.get_array_module(*inputs)
+          (x, w) = inputs
+
+          unless @ignore_label
+            return [Chainer::Utils::Array.take(w, x, axis: 0)]
+          end
+
+          mask = xm::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
```

Comment:
Is this equivalent with `mask = x.eq(@ignore_label)`? Avoid to_a if possible.

Reply:
I had looked for such a method! But I am not sure from [API reference](https://ruby-numo.github.io/narray/narray/Numo/Int16.html) that it is equivalent. 
I will try to check it.